docs/jsonapi-spec.md

```diff
@@ -1419,26 +1419,24 @@ The `$all` operation allows users to check for documents that have an array fiel
 
 If provided to a `find()`, return all the documents where the locations field contains the two values – “New York” and “Texas”. It does not matter how many more values the locations field contains. The operation will match if the specified values are present. Similarly, if even one the value is missing, the document will not be matched. So the `$all` operation is useful while retrieving data.  
 
-The `$all` is equivalent to an `$and` operation of the specified values. For example, this first statement:
+#### $elemMatch operation
 
-```json5
-{ tags: { $all: [ "ssl" , "security" ] } }
-```
+TODO:
 
-... is equivalent to:
+#### $size operation
 
-```json5
-{ $and: [ { tags: "ssl" }, { tags: "security" } ] }
+The `$size` operation allows users to match any array with the number of elements specified by the argument. Example:
 
+```json5
+db.collection.find( { field: { $size: 2 } } );
 ```
 
-#### $elemMatch operation
+That expression returns, e.g.: `{ field: [ red, green ] }` and `{ field: [ apple, lime ] }`, but not `{ field: fruit }` or `{ field: [ orange, lemon, grapefruit ] }`.
 
-TODO:
+If the given field is not an array, there's no match. `$size` should ignore non-arrays.
 
-#### $size operation
+Queries cannot use indexes for the $size portion of a query, although the other portions of a query can use indexes if applicable.
```

Comment:
Should have seen this before, $size index can be used along other filters and it uses the index. Can we remove this too?

Reply:
yes will do